### ISSUE TITLE
[Task]: Create a document that describes the current model / state of data integration code base

### ISSUE THREAD

**Issue by sumitsum** [appsmithorg/appsmith#23628]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

- This document is meant to serve as a reference for anyone who wants to get a high level understanding of the server and the client side of the code base.
- Current goal for creating this document is to server as a source of awareness as the data teams move forward to implement the strategy mentioned here. 
  - In short, before we can make the underlying system better, we need to understand what the system is. This document will serve to bring that understanding.
